tools/Import_Data_Normalise_Consent.php

```diff
@@ -0,0 +1,215 @@
+      foreach ($psData as $entry) {
+        if($entry[$consentName . '_date'] === "0000-00-00"){
+          array_push($errors, "Zero dates found in: " . $entry . ". Please remove date or run /tools/DB_date_zeros_removal.php.");
+        }
+      }
+  }
+}
+// Throw errors
+if (!empty($errors)) {
+  print_r($errors);
+  die("Resolve errors and run script again.\n");
+} else {
+    echo "\nValidation successful.\n";
+}
+
+// Continue script
+// Update ConfigSetting table with value of 'useConsent' if true. Default is set to false.
+$configID = $db->pselectOne(
+              'SELECT ID FROM ConfigSettings WHERE Name="useConsent"',
+              array()
+            );
+if ($useConsent === "true") {
+  $updateValue = ['Value' => $useConsent];
+  $db->update(
+       'Config',
+       $updateValue,
+       array (
+         'ConfigID' => $configID, 
+       )
+  );
+  echo "\nConfig settings set to use consent.\n\n";
+}
+
+// Start import of consent status information into new tables
+$consentType = [];
+$printArray  = [];
+
+foreach ($consents as $key=>$consent) {
+  $consentName  = $consent['name'];
+  $consentLabel = $consent['label'];
+  
+  // Populate consent_type table with consents from Config.xml
+  $db->insert(
+         'consent_type', 
+         array(
+           'Name'  => $consentName,
+           'Label' => $consentLabel,
+         )
+  );
+  // Save ConsentTypeID inserted
+  $consentTypeID = $db->pselectOne(
+                     'SELECT ConsentTypeID FROM consent_type WHERE Name=:consentName',
+                     array('consentName' => $consentName)
+                   );
+
+  // Create array of consents to use later in importing history data
+  $consentType[$consentName] = $consentLabel;
+
+  // Get all data where the consent status has a value
+  $psData = $db->pselect(
+              'SELECT * FROM participant_status WHERE '
+              . $consentName . ' IS NOT NULL OR ' . $consentName . ' != ""',
+              array()
+            );
+  foreach ($psData as $entry) {
+    // Push each formatted old entry to array
+    $consentValues = [
+        'CandidateID'   => $entry['CandID'],
+        'ConsentTypeID' => $consentTypeID,
+        'Status'        => $entry[$consentName],
+        'DateGiven'     => $entry[$consentName . '_date'],
+        'DateWithdrawn' => $entry[$consentName . '_withdrawal'],
+        ];
+    array_push($printArray, $consentValues);
+  }
+}
+
+// Output list of data to be inserted into new table to terminal
+echo "\nRows to be inserted into 'candidate_consent_type_rel' table ..\n";
+echo "Empty values will be inserted as NULL.\n\n";
+print_r($printArray);
+
+// Populate candidate_consent_type_rel with data
+foreach ($printArray as $consentValues) {
+    $db->insert('candidate_consent_type_rel', $consentValues);
+  }
+
+echo "\nData insert complete.\n";
+
+// Select consent history and import into new history table
+$consentHistory = $db->pselect(
+    'SELECT * FROM consent_info_history',
```

Comment:
Please specify the columns.

Reply:
i already said I AGREE !

Reply:
@xlecours thank you for your input!! i made the changes for all the other queries but for this one, the field names in the old history table are dynamically set by the consent types. I'll specify the predicted column names here but it'll be in a concatenated query